packages/api-server/api_server/repositories/tasks.py

```diff
@@ -27,6 +28,16 @@ def __init__(self, user: User):
         self.user = user
 
     async def save_task_state(self, task_state: TaskState) -> None:
+        datetime_request_time = (
+            datetime.fromtimestamp(task_state.unix_millis_request_time / 1000)
+            if task_state.unix_millis_request_time
+            else datetime.now()
+        )
+        task_state.unix_millis_request_time = (
+            task_state.unix_millis_request_time
+            if task_state.unix_millis_request_time
+            else round(time.time() * 1000)
+        )
```

Comment:
The above comment applies here as well, we will need to use `previous_task_state.unix_millis_request_time` instead

Reply:
I completely agree with you. 
But justly the value of `task_state.unix_millis_request_time` will be the previous value if the method is "update", and if it is "create" it will come null. 
So that's the control I'm doing there. If `task_state.unix_millis_request_time` has a value then keep using that value otherwise use datetime.now()

Reply:
yup, however we are still applying the keyword https://github.com/open-rmf/rmf-web/pull/689/files#diff-6e62b347b0bbff536cbf2a85cf360c53355a713c6f5bf02ef4aa1519b56db94bR55, which will always update with the values from `datetime_request_time`

request time will always increase whenever a new task state comes in,

![image](https://user-images.githubusercontent.com/5383623/225516060-5d0b9318-4dd5-4347-ba51-8858737c776c.png)

you can verify it by adding these 2 lines above [here](https://github.com/open-rmf/rmf-web/blob/feature/add-unix-millis-request-time/packages/api-server/api_server/repositories/tasks.py#L41), 
```python
        print('datetime_request_time: {}'.format(datetime_request_time))
        print('in json: {}'.format(task_state.booking.unix_millis_request_time))
```

Reply:
You are right!
Thanks for the thorough review.